src/routes/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
+import { IJwtService } from '@/datasources/jwt/jwt.service.interface';
+import { AuthService } from '@/routes/auth/auth.service';
+import { SiweMessageSchema } from '@/domain/auth/entities/siwe-message.entity';
+
+/**
+ * The AuthGuard should be used to protect routes that require authentication.
+ *
+ * It checks for the presence of a valid JWT access token in the request and
+ * verifies its validity before allowing access to the route.
+ *
+ * 1. Check for the presence of an access token in the request.
+ * 2. Verify the token's validity and decode it.
+ * 3. Ensure the decoded token is that of a valie SiweMessage.
```

Comment:
"valid"

Reply:
Fixed in b5aed5145112dbbc6648486f44d67c343fb43d7c.